### ISSUE TITLE
twisted.python.usage.Options errors when generic code for usage.Options definition has multiple underscores.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4858]
|<img alt="t_venkatesh's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| t_venkatesh reported|
|-|-|
|Trac ID|trac#4858|
|Type|defect|
|Created|2011-02-09 14:28:06Z|
|Branch|https://github.com/twisted/twisted/tree/underscore-options-4858|

When the generic code for options method has multiple underscores, and a short name is declared, twisted produces a traceback.

**To reproduce define:**
```
from twisted.python import usage
class TestOptions(usage.Options):
    def opt_verbose_big(self):
        "Call this multiple times to increment verbosity"
        self['verbosity'] += 1
    opt_v = opt_verbose_big
TestOptions().parseOptions()
# Invoke with -v
```

**Traceback:**
```
Traceback (most recent call last):
  File "/tmp/twistedOptions.py", line 29, in <module>
    TestOptions().parseOptions()
  File "<path>/twisted/python/usage.py", line 216, in parseOptions
    if isinstance(self._dispatch[optMangled], CoerceParameter):
KeyError: 'verbose_big'
```

**Patch:**
```
--- <path>/lib/python/twisted/python/usage.py        2011-02-09 08:35:08.000000000 -0500
+++ /tmp/usage.py       2011-02-09 08:35:02.000000000 -0500
@@ -393,5 +393,5 @@
             if method not in reverse_dct:
                 reverse_dct[method] = []
-            reverse_dct[method].append(name)
+            reverse_dct[method].append(name.replace('_','-'))

         cmpLength = lambda a, b: cmp(len(a), len(b))
```

Python 2.6.2

Twisted 8.2.0

<details><summary>Searchable metadata</summary>

```
trac-id__4858 4858
type__defect defect
reporter__t_venkatesh t_venkatesh
priority__normal normal
milestone__ 
branch__branches_underscore_options_4858 branches/underscore-options-4858
branch_author__t_venkatesh__exarkun t_venkatesh, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1297261686000000 1297261686000000
changetime__1297655884000000 1297655884000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [30648]) Branching to 'underscore-options-4858'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30649]) Apply the fix suggested in the ticket description, as well as a unit test.  Also document how Options works a little bit, and remove some old unittesty cruft.

refs [#4858](https://github.com/twisted/twisted/issues/4858)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @exarkun|
|-|-|

Looks great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [30750]) Merge underscore-options-4858 

Author: t_venkatesh, exarkun
Reviewer: glyph
Fixes: [#4858](https://github.com/twisted/twisted/issues/4858)

Fix a problem in `twisted.python.usage` handling options with a - in their
name which also have an alias which resulted in an unhandled exception
being raised when those options were parsed.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
